### PR TITLE
Update readme

### LINKED ISSUE #312: Add information in readme and documentation on installations of pyquil, qiskit for non devs
Right now, the readme and documentation simply refer to a development install for users interested in pyquil and qiskit. 

However, imagine that a non-dev user is interested (as a user, not as a contributor to mitiq) in using mitiq + pyquil or mitiq + qiskit, then he/she has no clear information on the version requirements for these libraries, i.e. from the `requirements.txt` 
```
# third-party integration
pyquil~=2.18.0
qiskit~=0.16.2
```
Such user would not be able to run the tests without these information. 

Proposed feature
-----------------
Add information about this in the readme and in the documentation, e.g., 

"""
#### Using mitiq with Qiskit

If you are willing to use mitiq to mitigate Qiskit circuits, make sure to have qiskit~=0.16.2 installed:
```
pip install qiskit~=0.16.2 mitiq
```

"""
Also, if we decide to do this, we could reference this back in the documentation's guide when such 3rd party executors are mentioned.

### PR DESCRIPTION
**Pull request by rmlarose**
Fixes #312 + the "spruce up readme" checklist todo I can't seem to find anymore.

Note I'm kinda guessing on which versions of Qiskit and pyQuil are supported. I know it works with Qiskit >= 0.19.0 and pyQuil >= 2.18.0, but earlier versions might be ok too.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,81 +6,70 @@
...
 
-## Contributing
-You can find information on contributing to `mitiq` code in the [contributing guidelines](CONTRIBUTING.md).
+## Developer information
 
-To contribute to the documentation, read the
-[instructions](docs/README-docs.md) in the `mitiq/docs` folder.
+We welcome contributions to Mitiq including bug fixes, feature requests, etc. Please see the 
+[contribution guidelines](CONTRIBUTING.md) for more details. To contribute to the documentation, please see these
+[documentation guidelines](docs/README-docs.md).
```
Comment (nathanshammah) [inline README.md:65]:
```suggestion
[documentation guidelines](CONTRIBUTING_DOCS).
```

**Comment by nathanshammah** [comment]
Thanks @rmlarose. 
The readme is converted as rst in the documentation and then gives [this page](https://mitiq.readthedocs.io/en/latest/README.html). In order for links to other (markdown) files in the repo to work also in the documentation, they need to be to the full URL link, not local, or they could point directly to the documentation version of such pages https://mitiq.readthedocs.io/en/latest/toc_contributing.html.

File: README.md [hunk 2]
```diff
@@ -6,81 +6,70 @@
...
+release can be found [here](https://mitiq.readthedocs.io/_/downloads/en/latest/pdf/).
 
-## Contributing
-You can find information on contributing to `mitiq` code in the [contributing guidelines](CONTRIBUTING.md).
+## Developer information
 
-To contribute to the documentation, read the
-[instructions](docs/README-docs.md) in the `mitiq/docs` folder.
+We welcome contributions to Mitiq including bug fixes, feature requests, etc. Please see the 
+[contribution guidelines](CONTRIBUTING.md) for more details. To contribute to the documentation, please see these
```
Comment (andreamari) [inline README.md:64]:
```suggestion
[contribution guidelines](https://github.com/unitaryfund/mitiq/blob/master/CONTRIBUTING.md) for more details. To contribute to the documentation, please see these
```

File: README.md (hunk 1 above)
Comment (andreamari) [inline README.md:65]:
```suggestion
[documentation guidelines](https://github.com/unitaryfund/mitiq/blob/master/docs/CONTRIBUTING_DOCS.md).
```

**Review by andreamari — APPROVED** [review]
LGTM! I suggested two changes which should fix the linking issue mentioned by @nathanshammah .

File: README.md (hunk 1 above)
Comment (rmlarose) [reply README.md]: Used the full url from Andrea's suggestion below.

**Pull request merged**